### ISSUE TITLE
HttpMethodMatcherPolicy.ApplyAsync ArgumentNullException 

### ISSUE THREAD

**Issue by spaasis** [dotnet/aspnetcore#39398]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

TL;DR: https://github.com/dotnet/aspnetcore/blob/077d0883e943bebbe8151ead202d4c18cc3bee6b/src/Http/Routing/src/Matching/HttpMethodMatcherPolicy.cs#L165 can throw ArgumentNullException (note the `!`) - the OrderBy should probably be skipped altogether if `methods` is null.

While tweaking my integration test infra to allow running the tests in parallel, I ran into a weird issue arising every now and then when running tests in a particular order. Not going into too much detail, but the issue reproduces if I run a specific set of tests in parallel. The issue can, weirdly enough, be circumvented by making a seemingly useless API call via the test server before running the actual test.

The stack trace:
```
System.ArgumentNullException
Value cannot be null. (Parameter 'source')
   at System.Linq.ThrowHelper.ThrowArgumentNullException(ExceptionArgument argument)
   at System.Linq.OrderedEnumerable`2..ctor(IEnumerable`1 source, Func`2 keySelector, IComparer`1 comparer, Boolean descending, OrderedEnumerable`1 parent)
   at System.Linq.Enumerable.OrderBy[TSource,TKey](IEnumerable`1 source, Func`2 keySelector, IComparer`1 comparer)
   at Microsoft.AspNetCore.Routing.Matching.HttpMethodMatcherPolicy.ApplyAsync(HttpContext httpContext, CandidateSet candidates)
   at Microsoft.AspNetCore.Routing.Matching.DfaMatcher.SelectEndpointWithPoliciesAsync(HttpContext httpContext, IEndpointSelectorPolicy[] policies, CandidateSet candidateSet)
   at Microsoft.AspNetCore.Routing.EndpointRoutingMiddleware.<Invoke>g__AwaitMatch|8_1(EndpointRoutingMiddleware middleware, HttpContext httpContext, Task matchTask)
   at Microsoft.AspNetCore.TestHost.HttpContextBuilder.<>c__DisplayClass23_0.<<SendAsync>g__RunRequestAsync|0>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.TestHost.ClientHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at Microsoft.AspNetCore.Mvc.Testing.Handlers.CookieContainerHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpClient.<SendAsync>g__Core|83_0(HttpRequestMessage request, HttpCompletionOption completionOption, CancellationTokenSource cts, Boolean disposeCts, CancellationTokenSource pendingRequestsCts, CancellationToken originalCancellationToken)
```

### Expected Behavior

The line of code in question should not expect `methods` to be not null

### .NET Version

6.0.100

**Comment by mkArtakMSFT** [comment]
Thanks for contacting us. we'd be happy if you'd send a PR for this.

**Comment by z77ma** [comment]
Please, excuse me if I broke a flow, since there is already a PR with a possible fix (closed atm, though)
